lib/FIR/mixin.h

```diff
@@ -25,6 +30,7 @@ namespace Fortran::FIR {
 
 inline constexpr bool has_size(std::size_t size) { return size > 0; }
 
+// implementation of a (moveable) sum type (variant)
 template<typename T, typename E = void> struct SumTypeMixin {};
 template<typename T>  // T must be std::variant<...>
```

Comment:
If T must be std::variant<>, it would be better to templatize on `typename... T` and then use `std::variant<T...>` below in place of bare T.